client-runtime/src/main/java/com/microsoft/rest/v2/policy/RequestPolicyChain.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.rest.v2.policy;
+
+import com.microsoft.rest.v2.http.HttpClient;
+import com.microsoft.rest.v2.http.HttpRequest;
+import com.microsoft.rest.v2.http.HttpResponse;
+import rx.Single;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Type representing request-response pipeline holding request policies.
+ */
+public class RequestPolicyChain extends HttpClient {
+    private final List<RequestPolicy.Factory> factories;
+
+    /**
+     * Creates RequestPolicyChain.
+     *
+     * @param factoryArray the factories that can creates RequestPolicy instances in the pipeline.
+     */
+    public RequestPolicyChain(RequestPolicy.Factory... factoryArray) {
+        factories = Arrays.asList(factoryArray);
```

Comment:
Chain this constructor into the one on line 40 so that you only have to use Collections.reverse() once.